hc_dpki/src/keypair.rs

```diff
@@ -1,610 +1,242 @@
-        let mut decrypted_data = SecBuf::with_secure(BUNDLE_DATA_LEN);
-        util::pw_dec(&data, passphrase, &mut decrypted_data, config)?;
-        let mut sign_priv = SecBuf::with_secure(SIGNATURESIZE);
-        let mut enc_priv = SecBuf::with_secure(32);
-
-        let pub_keys = {
-            let decrypted_data = decrypted_data.read_lock();
-            if decrypted_data[0] != 1 {
-                return Err(HolochainError::ErrorGeneric(format!(
-                    "Invalid Bundle Version : {:?}",
-                    decrypted_data[0]
-                )));
-            }
-            sign_priv.write(0, &decrypted_data[65..129])?;
-            enc_priv.write(0, &decrypted_data[129..161])?;
-
-            KeyBuffer::with_raw_parts(
-                array_ref![&decrypted_data, 1, 32],
-                array_ref![&decrypted_data, 33, 32],
-            )
-            .render()
-        };
-
-        Ok(Keypair {
-            pub_keys,
-            enc_priv,
-            sign_priv,
-        })
+    /// Construct with a public key not already HCID encoded
+    pub fn new_with_secbuf(pub_key_sec: &mut SecBuf, private: SecBuf) -> Self {
+        let public = Self::encode_pub_key(pub_key_sec);
+        Self { public, private }
     }
 
     /// sign some arbitrary data with the signing private key
-    ///
     /// @param {SecBuf} data - the data to sign
-    ///
-    /// @param {SecBuf} signature - Empty Buf the sign
-    pub fn sign(
-        &mut self,
-        data: &mut SecBuf,
-        signature: &mut SecBuf,
-    ) -> Result<(), HolochainError> {
-        sign::sign(data, &mut self.sign_priv, signature)?;
-        Ok(())
+    /// @return {SecBuf} signature - Empty SecBuf to be filled with the signature
+    pub fn sign(&mut self, data: &mut SecBuf) -> HcResult<SecBuf> {
+        let mut signature = SecBuf::with_insecure(SIGNATURE_SIZE);
+        holochain_sodium::sign::sign(data, &mut self.private, &mut signature)?;
+        Ok(signature)
     }
 
     /// verify data that was signed with our private signing key
-    ///
-    /// @param {SecBuf} signature
-    ///
     /// @param {SecBuf} data
-    pub fn verify(
-        pub_keys: String,
-        signature: &mut SecBuf,
-        data: &mut SecBuf,
-    ) -> Result<i32, HolochainError> {
-        let mut sign_pub = SecBuf::with_insecure(sign::PUBLICKEYBYTES);
-        let mut enc_pub = SecBuf::with_insecure(kx::PUBLICKEYBYTES);
-
-        util::decode_id(pub_keys, &mut sign_pub, &mut enc_pub)?;
-        let verified: i32 = sign::verify(signature, data, &mut sign_pub);
-        Ok(verified)
+    /// @param {SecBuf} signature
+    /// @return true if verification succeeded
+    pub fn verify(&mut self, data: &mut SecBuf, signature: &mut SecBuf) -> bool {
+        let mut pub_key = self.decode_pub_key_into_secbuf();
+        let res = holochain_sodium::sign::verify(signature, data, &mut pub_key);
+        res == 0
```

Comment:
can we make the base `holochain_sodium::sign::verify` return the boolean?

Reply:
Done